scst-scan/custom-ivs-samples.md

```diff
@@ -66,3 +66,85 @@ Where:
   - `scanner` is the service account that runs the scan. It must have read access to `image`.
   - `publisher` is the service account that uploads results. It must have write access to `scanResults.location`.
 - `CARBON-BLACK-SCANNER-IMAGE` is the Carbon Black Scanner image used to run Carbon Black scans.
+
+### Configure a ImageVulnerabilityScan for Prisma
+
+1. Have an image with the following:
+     - [podman](https://podman.io/docs/installation)
+     - [twistcli](https://docs.paloaltonetworks.com/prisma/prisma-cloud/prisma-cloud-admin-compute/tools/twistcli)
+2. Using the yaml below, populate the placeholder fields with the appropriate values and save to a file `ivs-prisma.yaml`:
+    ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: prisma-auth
+      namespace: app-scanning
+    stringData:
+      username: USERNAME
+      password: PASSWORD
+      address: ADDRESS
+    ---
+    apiVersion: app-scanning.apps.tanzu.vmware.com/v1alpha1
+    kind: ImageVulnerabilityScan
+    metadata:
+      name: prisma-scan
+      namespace: app-scanning
+    spec:
+      image: harbor-repo.vmware.com/dockerhub-proxy-cache/library/nginx@sha256:6650513efd1d27c1f8a5351cbd33edf85cc7e0d9d0fcb4ffb23d8fa89b601ba8
+      scanResults:
+        location: harbor-repo.vmware.com/scst_scan_reports/prisma-results
+      serviceAccountNames:
+        publisher: report-publisher
+        scanner: scanner
+      steps:
+      - name: prisma
+        image: PRISMA-SCANNER-IMAGE
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+          runAsUser: 0
```

Comment:
```
privileged: true
runAsUser: 0
```
These two lines are needed to do this twistcli scan:

These were the errors I was getting:

without securityContext.privileged: true
```
+ ./twistcli images scan --podman-path /usr/bin/podman --address https://us-west1.cloud.twistlock.com/us-4-161033339 --project 'Central Console' --user 2890ea9b-04cf-4753-9dbe-2f530108dc40 --password bYkxYmy1ziOk1OLoJIIs7E/npYg= harbor-repo.vmware.com/dockerhub-proxy-cache/library/nginx@sha256:6650513efd1d27c1f8a5351cbd33edf85cc7e0d9d0fcb4ffb23d8fa89b601ba8 --output-file workspace/twist-scan.json
failed to augment data: exit status 125
```

without securityContext.runAsUser: 0
```
robin@lrobin6MD6R app-scanning % kubectl -n app-scanning logs prisma-scan-mcgbx-scan-task-pod -c step-prisma
time="2023-06-08T20:13:09Z" level=warning msg="\"/\" is not a shared mount, this could cause issues or missing mounts with rootless containers"
time="2023-06-08T20:13:09Z" level=error msg="cannot find UID/GID for user nobody: open /etc/subuid: no such file or directory - check rootless mode in man pages."
time="2023-06-08T20:13:09Z" level=warning msg="Using rootless single mapping into the namespace. This might break some images. Check /etc/subuid and /etc/subgid for adding sub*ids if not using a network user"
Trying to pull harbor-repo.vmware.com/dockerhub-proxy-cache/library/nginx@sha256:6650513efd1d27c1f8a5351cbd33edf85cc7e0d9d0fcb4ffb23d8fa89b601ba8...
Getting image source signatures
Copying blob sha256:7e9b29976cce358fbdd777b91ce2c52d18090dd1507d0616793e9ce8cc15979c
Copying blob sha256:077b9569ff86b205b47334ce8fa1c4bcf55b124b2fe880f40b56f9932f856f68
Copying blob sha256:bb263680fed18eecdc67f885094df6f589bafc19004839d7fdf141df236a61aa
Copying blob sha256:3082a16f3b619613a40d0b9056c6142493980314187ef6b6281909a40f6b7147
Copying blob sha256:258f176fd22609802ed5def498ce4c7d1fa80a8ddba258b6e9eca0364df3ddb3
Copying blob sha256:a0bc35e7077319c74cd587e1c911894c0bc003e1783c0bc2e59e351d41cfa543
Error: copying system image from manifest list: writing blob: adding layer with blob "sha256:bb263680fed18eecdc67f885094df6f589bafc19004839d7fdf141df236a61aa": processing tar file(potentially insufficient UIDs or GIDs available in user namespace (requested 0:42 for /etc/gshadow): Check /etc/subuid and /etc/subgid if configured locally and run podman-system-migrate: lchown /etc/gshadow: invalid argument): exit status 1
failed to find image harbor-repo.vmware.com/dockerhub-proxy-cache/library/nginx@sha256:6650513efd1d27c1f8a5351cbd33edf85cc7e0d9d0fcb4ffb23d8fa89b601ba8
```

I'm not sure about having this here. Any ideas? Should I leave this out and put a note about this?

Reply:
Is this the only way to resolve this?  Feels like maybe we hit a nail with a sledge hammer.  I don't think we should be providing guidance to run a container as root and in privileged mode.  Any idea how labs got around this in their implementation for scan-link?